examples/custom/gears/gear.ts

```diff
@@ -142,7 +142,7 @@ export class GearView extends XYGlyphView {
   }
 
   draw_legend_for_index(ctx: Context2d, bbox: Rect, index: number): void {
-    generic_area_legend(this.visuals, ctx, bbox, index)
+    generic_area_vector_legend(this.visuals, ctx, bbox, index)
```

Comment:
Could be not avoid this breakage but having the common (vector) case remain `generic_area_legend` (etc) and the uncommon scalar case be `generic_area_legend_scalar`.

Otherwise, if we are going to break this we need a release note.

Reply:
I added a backwards compatibility layer.